src/olympia/amo/middleware.py

```diff
@@ -248,7 +250,9 @@ def process_exception(self, request, exception):
         if isinstance(exception, mysql.OperationalError):
             if request.is_api:
                 return self._render_api_error()
-            return render(request, 'amo/read-only.html', status=503)
+            response = render(request, 'amo/read-only.html', status=503)
+            response.render()  # Force render to return an HttpResponse
+            return response
```

Comment:
if you were in the mood for refactoring, these lines are a bit DRYy with `process_request`

Reply:
https://github.com/mozilla/addons-server/pull/18295/commits/20b612358178e8fb6d5a55212327c8c69f06e52b